File: app/backend/java/smartride-ai-service/src/main/java/com/smartride/smartride_ai_service/dto/ChatRequest.java
package com.smartride.smartride_ai_service.dto;

import com.fasterxml.jackson.annotation.JsonAlias;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

/**
 * Request DTO for the AI chat endpoint.
 *
 * message   - Required. The user's question or message.
 * sessionId - Optional. UUID for conversation context (NOT used for auth).
 */
public class ChatRequest {

    @NotBlank(message = "Message must not be blank")
    @Size(max = 2000, message = "Message must not exceed 2000 characters")
    @JsonAlias({"prompt", "text", "query", "userMessage"})
    private String message;

    private String sessionId; // optional — for conversation context only

    // --- Getters & Setters ---

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public String getSessionId() {
        return sessionId;
    }

    public void setSessionId(String sessionId) {
        this.sessionId = sessionId;
    }
}

File: app/backend/.net/SmartRide.AiService/Models/ChatRequest.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace SmartRide.AiService.Models;

/// <summary>
/// Request payload for the SmartRide AI Chat endpoint.
/// </summary>
public class ChatRequest
{
    /// <summary>
    /// The primary user message. Accepts aliases: 'prompt', 'text', 'query', 'userMessage'.
    /// </summary>
    [Required(ErrorMessage = "Message is required.")]
    [MaxLength(2000, ErrorMessage = "Message must not exceed 2000 characters.")]
    [JsonPropertyName("message")]
    public string Message { get; set; } = string.Empty;

    /// <summary>
    /// Optional session UUID for conversation context.
    /// </summary>
    [JsonPropertyName("sessionId")]
    public string? SessionId { get; set; }

    // Fallback Property Aliases for flexible JSON binding
    [JsonPropertyName("prompt")]
    public string? PromptAlias { set { if (!string.IsNullOrWhiteSpace(value) && string.IsNullOrWhiteSpace(Message)) Message = value; } }

    [JsonPropertyName("text")]
    public string? TextAlias { set { if (!string.IsNullOrWhiteSpace(value) && string.IsNullOrWhiteSpace(Message)) Message = value; } }

    [JsonPropertyName("query")]
    public string? QueryAlias { set { if (!string.IsNullOrWhiteSpace(value) && string.IsNullOrWhiteSpace(Message)) Message = value; } }

    [JsonPropertyName("userMessage")]
    public string? UserMessageAlias { set { if (!string.IsNullOrWhiteSpace(value) && string.IsNullOrWhiteSpace(Message)) Message = value; } }
}
